### PR TITLE
#400 Junit5 parametrized

### PR DESCRIPTION
**Pull request by HDouss**
Resolving issue #400

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#404](https://github.com/yegor256/jpeek/issues/404) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/78f310ad-5adf-4135-b2fb-be3d58b1a536); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by HDouss** [comment]
@fanifieiev ping

File: src/test/java/org/jpeek/MetricsTest.java [hunk 1]
```diff
@@ -55,229 +55,216 @@
...
+                    target, value
                 )
             )
         ).affirm();
     }
+
+    // @checkstyle MethodLengthCheck (200 lines)
+    private static Collection<Arguments> targets() {
+        return new CollectionOf<>(
+            Arguments.of("NoMethods", "NHD", Double.NaN),
```
Comment (fanifieiev) [inline src/test/java/org/jpeek/MetricsTest.java]: @HDouss This method looks huge and it does load parameters.
So, why not creating a file with a format:
```
Bar, LCOM, 6.0d
Foo, LCOM, 1.0d
.....
```
and use the folllowing:
```
final List<Arguments> lines = new Mapped<>(
            line -> Arguments.of(
                new org.cactoos.iterable.Mapped<>(
                    text -> text.asString(),
                    new Split(
                        line, ","
                    )
                )
            ),
            new Split(
                new TextOf(
                    new ResourceOf("path_to_file")
                ),
                "\n"
            )
        );
```

File: src/test/java/org/jpeek/MetricsTest.java [hunk 2]
```diff
@@ -55,229 +55,216 @@
...
-            new Object[] {"TwoCommonMethods", "OCC", 0.0d},
-            new Object[] {"WithoutAttributes", "OCC", 0.0d},
-            new Object[] {"TwoCommonMethods", "LORM", 0.26_667d}
-        );
-    }
-
-    @Test
-    public void testsTarget() throws Exception {
+    @ParameterizedTest
+    @MethodSource("targets")
```
Comment (fanifieiev) [inline src/test/java/org/jpeek/MetricsTest.java]: @HDouss Even though it is a feature of Junit5, I find this kind of features very confusing. Would personally not use.
@paulodamaso WDYT???

Comment (HDouss) [reply src/test/java/org/jpeek/MetricsTest.java]: @fanifieiev I think it still better than Junit 4 way where you return an `Object` array and then get mapped by indices to instance fields. Anyway, I think my last commit propose an approach that adresses all your comments.

**Comment by HDouss** [comment]
@fanifieiev Thanks for your review, please see commit 797f37e

**Comment by HDouss** [comment]
@paulodamaso ping

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)

**Pull request merged**